distributed/worker.py

```diff
@@ -161,8 +160,8 @@ def heartbeat(self):
             self.heartbeat_active = True
             logger.debug("Heartbeat: %s" % self.address)
             try:
-                if proc:
-                    memory_info = proc.memory_info()
+                if psutil:
+                    memory_info = psutil.Process().memory_info()
```

Comment:
I was worried that it might be slower on other platforms but it seems fine on Windows as well:
```
>>> %timeit _ = psutil.Process()
The slowest run took 11.32 times longer than the fastest. This could mean that a
n intermediate result is being cached.
100000 loops, best of 3: 6.76 µs per loop
```